llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Passes/FoldIntoPackAndUnpackOps.cpp

```diff
@@ -26,6 +26,34 @@ static bool areAllConstantIntValue(ArrayRef<OpFoldResult> ofrs, int64_t value) {
       ofrs, [&](OpFoldResult ofr) { return isConstantIntValue(ofr, value); });
 }
 
+/// Fold a `pad` -> `pack` into `pack` if they have the same padding values and
+/// the pad op has zero low paddings.
+struct FoldPadWithPackOp : public OpRewritePattern<PackOp> {
+  using OpRewritePattern<PackOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(PackOp packOp,
+                                PatternRewriter &rewriter) const override {
+    auto padOp = packOp.getInput().getDefiningOp<tensor::PadOp>();
+
+    if (!padOp || padOp.getNofold() || !padOp.hasZeroLowPad())
```

Comment:
I knew I had done something similar to this somewhere. Found it [here](https://github.com/iree-org/iree/blob/533c2464dc0cc5429a57b3c22a69826007f91a08/llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Passes/MaterializeEncoding.cpp#L107). I am not sure we want to unify it, its not too much code. More an FYI.